typescript/introduction/tsc-data-types-ii/tsc-object-type-comparison.md

```diff
@@ -0,0 +1,146 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# `Object` vs `object` vs `{}`
+
+---
+## Content
+
+TypeScript has 3 ways to represent an object with slightly different behaviors.
+
+Since any defined value in JavaScript is an object[1], the `Object` type can be used to represent any type that is considered to contain a value:
+
+```ts
+let num: Object = 3;
+let str: Object = 'enki';
+let arr: Object = [1, 2, 3];
+// ...
+// only null and undefined give an error
+let n: Object = null; // error
+let u: Object = undefined; // error
+```
+
+This often leads to confusion. Usually by "object" we only mean collections of values (e.g. maps, sets, arrays, plain objects), but not primitive values[2] (e.g. numbers or strings).
+
+This is where the `object` type comes into play.
+
+```ts
+let arr: object = [1, 2, 3];
+let obj: object = { enki: true };
+let map: object = new Map([["enki", "cool"]])
+// ...
+// primitive values give an error
+let str: object = 'enki'; // error
+let num: object = 3; // error
+let n: object = null; // error
+let u: object = undefined; // error
+```
+
+The type `{}` is used to represent a JavaScript object with a specific shape.
```

Comment:
I'd personally split this into 2 insights:
- first insight about `{}` and how to define standard objects with specific shapes; this is the one mostly used in TS I think
- 2nd insight about `object` and `Object` where we present them like special types similar to `never` `any` etc

Reply:
this will allow us to add more granual practice/testing opportunites